### ISSUE TITLE
`Style/RedundantBegin` and lambda

### ISSUE THREAD

**Issue by marocchino** [rubocop/rubocop#10705]
Hi.

I found something that looked like a bug and came to report it.

--------

## Expected behavior

should not change below codes.
```rb
lambda do |slug|
  begin
    joins(:groups).where(groups: { id: Group.find_by!(slug:) })
  rescue ActiveRecord::RecordNotFound
    none
  end
end
```

## Actual behavior

it is reported as `Style/RedundantBegin` and auto corrected with
```rb
lambda do |slug|
  joins(:groups).where(groups: { id: Group.find_by!(slug:) })
rescue ActiveRecord::RecordNotFound
  none
end
```
It is cause ArgumentError
```
ArgumentError:
  tried to create Proc object without a block
```
## Steps to reproduce the problem

Just rescue it in the lambda of the do end block and it will be reproduced.

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler.
If you see extension cop versions (e.g. `rubocop-performance`, `rubocop-rspec`, and others)
output by `rubocop -V`, include them as well. Here's an example:

```
$ rubocop -V
1.30.1 (using Parser 3.1.2.0, rubocop-ast 1.18.0, running on ruby 3.1.2 arm64-darwin21)
  - rubocop-performance 1.14.2
  - rubocop-rails 2.14.2
  - rubocop-rspec 2.11.1
  - rubocop-rubycw 0.1.6
```

**Comment by koic** [comment]
Hm, It looks like the `begin` can be omitted.

```ruby
x = lambda do |bool|
  if bool
    'foo'
  else
    raise
  end
rescue
  'bar'
end

puts x.call(true)  # foo
puts x.call(false) # bar
```

Can you provide the `ArgumentError: tried to create Proc object without a block` error reproduction (expected and actual) code in PORO that does not depend on Active Record?

**Comment by marocchino** [comment]
aha.
It was about the interpretation of the do-end block.

So did not pass, but B passed.

However, when it was C, there was no problem, so it can not be helped that it seems to be broken by auto-complete.

```rb
```

**Comment by marocchino** [comment]
Thank you for your reproduce.
And... You are right.
It was about the interpretation of the do-end block.

So 
```rb
scope :group_slug, lambda do |slug|
  joins(:groups).where(groups: { id: Group.find_by!(slug:) })
rescue ActiveRecord::RecordNotFound
  none
end
```
did not pass, but
```rb
scope(:group_slug, lambda do |slug|
  joins(:groups).where(groups: { id: Group.find_by!(slug:) })
rescue ActiveRecord::RecordNotFound
  none
end)
```

passed.

However, when it was 
```
scope :group_slug, lambda do |slug|
  begin
    joins(:groups).where(groups: { id: Group.find_by!(slug:) })
  rescue ActiveRecord::RecordNotFound
    none
  end
end
```
there was no problem, so it can not be helped that it seems to be broken by auto-complete.
